src/test/resources/connections.properties

```diff
@@ -5,14 +5,18 @@
 # the ports specified in the JDBC URLs are the default ones and are used by the Timeouts test
 ###############################################################################################
 
+# -=- H2 LEGACY MODE -=-
+# Version 2.x of H2 is not compatible with version 1.x. When the version was bumped to 2.x,
+# the H2 legacy mode was turned on to keep some backward compatibility with the 1.x version.
+# For more information, see: http://www.h2database.com/html/features.html.
 
 h2.engine=com.feedzai.commons.sql.abstraction.engine.impl.H2Engine
-h2.jdbc=jdbc:h2:./target/pdb;AUTO_SERVER=FALSE
+h2.jdbc=jdbc:h2:./target/pdb;AUTO_SERVER=FALSE;MODE=LEGACY
```

Comment:
This doesn't seem very bad for now, but in which ways is v2.x not compatible with v1.x?
Would it be too much trouble creating an implementation of H2Enginev2?

Reply:
The migration changes are documented here: https://www.h2database.com/html/migration-to-v2.html

I don't think that is necessary. I did tests with CM and Genome so far and none of them need the legacy mode. The legacy mode here in PDB is mainly because there are several tests that would fail without that (mainly because the way that H2 handled the ID counter in auto-increment columns, which changed in the new version). Fixing these tests is not trivial and that is why I opted to use the legacy mode.